### PR TITLE
`VaSelect` updates and refactoring

### PR DESCRIPTION
**Pull request by aluarius**
Close: #2673 
Close: #2795 

## Description
- [x] added `max-visible-options` prop to limit the output inside of input wrapper;
- [x] added `selected-top-shown` prop to show selected options at the top of the options list;
- [x] refactoring. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-select/VaSelect.vue [hunk 1]
```diff
@@ -168,11 +189,15 @@ import {
 import { extractComponentProps, filterComponentProps } from '../../utils/component-options'
 
 import { VaDropdown, VaDropdownContent } from '../va-dropdown'
+import { VaBadge } from '../va-badge'
 import { VaIcon } from '../va-icon'
 import { VaInput, VaInputWrapper } from '../va-input'
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-select/VaSelect.vue:194]:
```suggestion
import {
  VaDropdown,
  VaDropdownContent,
  VaBadge,
  VaIcon,
  VaInput,
  VaInputWrapper,
} from '../index'
```

File: packages/ui/src/components/va-select/hooks/useMaxVisibleOptions.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { computed, ref, toRefs, watch, ExtractPropTypes } from 'vue'
+
+import type { SelectOption } from '../types'
+
+export const useMaxVisibleOptionsProps = {
+  maxVisibleOptions: { type: Number, default: 0 },
+}
+
+export const useMaxVisibleOptions = (
+  props: ExtractPropTypes<typeof useMaxVisibleOptionsProps> & { multiple: boolean, modelValue: SelectOption | SelectOption[] },
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-select/hooks/useMaxVisibleOptions.ts:10]: Imo, will be better to move to the separate type and use it here.

File: packages/ui/src/components/va-select/hooks/useMaxVisibleOptions.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+) => {
+  const { modelValue } = toRefs(props)
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-select/hooks/useMaxVisibleOptions.ts:12]: If here no plans to make another props reactive, then will be great to use `toRef()`.

File: packages/docs/src/locales/en/en.json [hunk 4]
```diff
@@ -3238,6 +3244,14 @@
       "styles": {
         "title": "Styles",
         "text": "Select styles inherit from input styles"
+      },
+      "maxVisibleOptions": {
+        "title": "Max visible selected options amount",
```
Comment (rustem-nasyrov) [inline packages/docs/src/locales/en/en.json:3249]:
```suggestion
        "title": "Max visible selected options",
```

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]
Great! But here a few suggestions.

**Comment by rustem-nasyrov** [comment]
image
Can we improve this?

**Comment by rustem-nasyrov** [comment]
Also, would be great to have demos in the vue-book.

**Comment by aluarius** [comment]
> Also, would be great to have demos in the vue-book.

We already have it:

image

**Comment by aluarius** [comment]
> Can we improve this?

Will implement nowrap class for this case.

**Review by rustem-nasyrov — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]

**Pull request merged**